src/d3a/models/market/market_structures.py

```diff
@@ -85,6 +97,27 @@ def _to_csv(self):
         rate = round(self.price / self.energy, 4)
         return self.id, rate, self.energy, self.price, self.buyer
 
+    def to_JSON_string(self):
+        return json.dumps(self._asdict())
+
+
+def bid_from_JSON_string(bid_string):
+    bid_dict = json.loads(bid_string)
+    return Bid(**bid_dict)
+
+
+class TradeBidInfo(namedtuple('TradeBidInfo',
+                              ('original_bid_rate', 'propagated_bid_rate',
+                               'original_offer_rate', 'propagated_offer_rate',
+                               'trade_rate'))):
+    def to_JSON_string(self):
+        return json.dumps(self._asdict())
+
+
+def trade_bid_info_from_JSON_string(info_string):
+    info_dict = json.loads(info_string)
+    return Bid(**info_dict)
```

Comment:
Please exchange `Bid` with `TradeBidInfo` Thanks!

Reply:
Oops, thanks a lot. Fixing this